### ISSUE TITLE
Make sure `create_dunedaq_package` on the `develop` branch of `daq-cmake` works for v5

### ISSUE THREAD

**Issue by jcfreeman2** [DUNE-DAQ/daq-cmake#136]
In a nutshell: I wrote `create_dunedaq_package` two summers ago, and it's quite possible that, e.g., the DAQ modules it generates are no longer compatible with the cutting-edge v5 code we work with. This should be checked, and if there's anything broken, it should be fixed.

**Comment by jcfreeman2** [comment]
A test Andrew described to me on Slack:

"""
Looks like create_dunedaq_package does not work out of the box in v5.
```
source /cvmfs/dunedaq.opensciencegrid.org/setup_dunedaq.sh
setup_dbt latest_v5
dbt-create -n NFD_DEV_240702_A9 test_create_dunedaq_package_v5
cd test_create_dunedaq_package_v5 && . env.sh
cd sourcecode/
create_dunedaq_package --daq-module AFirstModule --config-generation --user-app an_app_for_users --user-app another_app_for_users --python-bindings --main-library thenewpackage
cd ..
dbt-workarea-env
dbt-build -j 24
```

On building, I got these errors:

```
In file included from /home/nfs/amogan/test_create_dunedaq_package_v5/sourcecode/thenewpackage/plugins/AFirstModule.cpp:11:
/home/nfs/amogan/test_create_dunedaq_package_v5/sourcecode/thenewpackage/plugins/AFirstModule.hpp:27:8: error: 'void dunedaq::thenewpackage::AFirstModule::init(const dunedaq::appfwk::DAQModule::data_t&)' marked 'override', but does not override
   27 |   void init(const data_t&) override;
      |        ^~~~
In file included from /home/nfs/amogan/test_create_dunedaq_package_v5/sourcecode/thenewpackage/plugins/AFirstModule.hpp:14:
/home/nfs/amogan/test_create_dunedaq_package_v5/sourcecode/thenewpackage/plugins/AFirstModule.cpp: In function 'std::shared_ptr<dunedaq::appfwk::DAQModule> make(std::string)':
/home/nfs/amogan/test_create_dunedaq_package_v5/sourcecode/thenewpackage/plugins/AFirstModule.cpp:49:1: error: invalid new-expression of abstract class type 'dunedaq::thenewpackage::AFirstModule'
   49 | DEFINE_DUNE_DAQ_MODULE(dunedaq::thenewpackage::AFirstModule)
      | ^~~~~~~~~~~~~~~~~~~~~~
/home/nfs/amogan/test_create_dunedaq_package_v5/sourcecode/thenewpackage/plugins/AFirstModule.hpp:22:7: note:   because the following virtual functions are pure within 'dunedaq::thenewpackage::AFirstModule':
   22 | class AFirstModule : public dunedaq::appfwk::DAQModule
      |       ^~~~~~~~~~~~
/cvmfs/dunedaq-development.opensciencegrid.org/nightly/NB_DEV_240702_A9/spack-0.20.0/opt/spack/linux-almalinux9-x86_64/gcc-12.1.0/appfwk-NB_DEV_240702_A9-uij2jpi7y4klxi4nieos3b7lhchusw4c/include/appfwk/DAQModule.hpp:187:16: note:     'virtual void dunedaq::appfwk::DAQModule::init(std::shared_ptr<dunedaq::appfwk::ModuleConfiguration>)'
  187 |   virtual void init(std::shared_ptr<ModuleConfiguration> /*mcfg*/) = 0;
      |                ^~~~
ninja: build stopped: subcommand failed.
[9:10](https://dunescience.slack.com/archives/D06EZLJDH3P/p1719929445208339)
```

Repeating those same steps with v4 and PROD4 works
"""

**Comment by jcfreeman2** [comment]
Fixed with PR #137 about six weeks ago.
